Newsletters: add 235 (2023-01-25)

harding opened the pull request:
- [ ] Releases/RCs and topic links @harding 
- [x] Bitcoin Stack Exchange @bitschmidty 
- [x] Bitcoin Core 26325 @Xekyo

Xekyo reviewed (commented):
LGTM

LarryRuane commented on _posts/en/newsletters/2023-01-25-newsletter.md:
```diff
@@ -0,0 +1,121 @@
+
+    Ephemeral anchors, similar to [fee sponsorship][topic fee
+    sponsorship], lets anyone [CPFP][topic cpfp] fee bump a transaction.
+    The transaction being fee bumped is allowed to contain zero fees.
```
```suggestion
    The transaction being fee-bumped is allowed to contain zero fees.
```

LarryRuane commented on _posts/en/newsletters/2023-01-25-newsletter.md:
```diff
@@ -0,0 +1,121 @@
+    this mechanism can also be used to pay fees for the multi-input
+    presigned transactions which are a target for `SIGHASH_GROUP`.
+
+    `SIGHASH_GROUP` would still have two advantages: first, that it
```
```suggestion
    `SIGHASH_GROUP` would still have two advantages: first, it
```

LarryRuane commented on _posts/en/newsletters/2023-01-25-newsletter.md:
```diff
@@ -0,0 +1,121 @@
+    `SIGHASH_GROUP` would still have two advantages: first, that it
+    could allow [batching][topic payment batching] multiple unrelated
+    presigned transactions, which could reduce transaction size
+    overhead, which would reduce user costs and increase network
```
```suggestion
    overhead, reducing user costs and increasing network
```

LarryRuane commented on _posts/en/newsletters/2023-01-25-newsletter.md:
```diff
@@ -0,0 +1,121 @@
+    could allow [batching][topic payment batching] multiple unrelated
+    presigned transactions, which could reduce transaction size
+    overhead, which would reduce user costs and increase network
+    capacity.  Second, that it doesn't require a child transaction,
```
```suggestion
    capacity.  Second, it doesn't require a child transaction,
```

LarryRuane commented on _posts/en/newsletters/2023-01-25-newsletter.md:
```diff
@@ -0,0 +1,121 @@
+    which would further reduce costs and increase capacity.
+
+    Towns concludes by noting that ephemeral anchors, with its
+    dependency of [v3 transaction relay][topic v3 transaction relay],
```
```suggestion
    dependency on [v3 transaction relay][topic v3 transaction relay],
```

LarryRuane commented on _posts/en/newsletters/2023-01-25-newsletter.md:
```diff
@@ -0,0 +1,115 @@
+
+    Towns concludes by noting that ephemeral anchors, with its
+    dependency of [v3 transaction relay][topic v3 transaction relay],
+    capture most of the benefits of `SIGHASH_GROUP` and provides the
```
I'd say your first suggestion is correct since the subject is singular -- "ephemeral anchors", even though it ends in "s", is singular because it's a concept. I considered suggesting "the ephemeral anchors concept" (or "approach") on line 49 but I think that's too wordy.

LarryRuane commented on _posts/en/newsletters/2023-01-25-newsletter.md:
```diff
@@ -0,0 +1,121 @@
+Proposals (BIPs)][bips repo], and [Lightning BOLTs][bolts repo].*
+
+- [Bitcoin Core #26325][] improves the results of the `scanblocks` rpc
+  by removing false positives in a second pass. `scanblock` may be used
```
```suggestion
  by removing false positives in a second pass. `scanblocks` may be used
```

LarryRuane commented on _posts/en/newsletters/2023-01-25-newsletter.md:
```diff
@@ -0,0 +1,121 @@
+
+- [Bitcoin Core #26325][] improves the results of the `scanblocks` rpc
+  by removing false positives in a second pass. `scanblock` may be used
+  to find blocks transactions relevant to a provided set of descriptors.
```
```suggestion
  to find blocks containing transactions relevant to a provided set of descriptors.
```

LarryRuane commented on _posts/en/newsletters/2023-01-25-newsletter.md:
```diff
@@ -0,0 +1,121 @@
+  Since scanning against the filters may falsely indicate some blocks
+  that do not actually contain relevant transactions, this PR validates
+  each hit in another pass to see whether blocks actually correspond to
+  the passed descriptors before providing the results to the caller.
```
It may be worth mentioning that this functionality is disabled by default (for performance reasons); the user must specify `-filter_false_positives` to enable it.

LarryRuane reviewed (commented):
I verified the SE question numbers, and lightly reviewed the content of the summaries.

bitschmidty commented on _posts/en/newsletters/2023-01-25-newsletter.md:
```diff
@@ -0,0 +1,167 @@
+  builds] build attestations.
+
+- [Why doesn't signet use a unique bech32 prefix?]({{bse}}116630)
+  Casey Rodarmor wonders why both testnet and [signet][topic signet] both use
```
Thanks Larry, took this suggestion

bitschmidty commented on _posts/en/newsletters/2023-01-25-newsletter.md:
```diff
@@ -0,0 +1,167 @@
+  RedGrittyBrick points to [one of several][large witness tx] recent P2TR
+  transactions containing a large amount of witness data. Other users point out
+  that the Ordinals project provides a service for including arbitrary data,
+  like [the image][ordinals example] in the transaction above, into a Bitcoin transaction using the witness.
```
I didn’t think the response was notable (as far as sipa answers go anyways!), personally. Other than noting it was a P2TR, which I did add for context. Which part do you suggest?

LarryRuane replied:
Okay; I thought you might not be aware that he had answered, since it was so recent.

I thought it was interesting (and new to me) that P2TR spends' standardness rules are "much more relaxed" than (at least) segwit v0 rules. But now that I read his answer again, he is not sure if the transactions the question is about are considered standard or not. So you're right, not worth mentioning here. 

I also thought that the summaries here _always_ include at least something from the answers given, not just the question. But there's no reason we can't just highlight a question and let readers follow the link if they want to know more. So, in short, never mind, it's okay as is.

bitschmidty replied:
Yeah, there isnt a great answer yet on that one but I thought it was interesting.

Also found per their docs: 

> Inscription content is included in transaction witnesses, which receive the witness discount. To calculate the approximate fee that in inscribe transaction will pay, divide the content size by four and muliply by the fee rate.

> Inscription transactions must be less than 400,000 weight units, or they will not be relayed by Bitcoin Core. One byte of inscription content costs one weight unit. Since an inscription transaction includes not just the inscription content, limit inscription content to less than 400,000 weight units. 390,000 weight units should be safe.

https://docs.ordinals.com/guides/inscriptions.html

So it sounds like it is relayed under 400,000wu

harding commented on _posts/en/newsletters/2023-01-25-newsletter.md:
```diff
@@ -0,0 +1,121 @@
+  later when an appropriate feerate is known, and the existing
+  `SIGHASH_ANYONECANPAY` and `SIGHASH_SINGLE` sighash flags aren't
+  flexible enough for multi-input transactions because they only commit
+  to a single input or output.
```
I think or is more correct, as `SIGHASH_SINGLE` signs multiple outputs (unless a transaction only has one output).

LarryRuane replied:
Are you sure? Antonop's book (table 4 of the section I linked to above) says
> SINGLE - Signature applies to all inputs but only the one output with the same index number as the signed input
> SINGLE|ANYONECANPAY - Signature applies to one input and the output with the same index number

harding replied:
Sorry, I meant it signs multiple inputs.

harding commented:
Made edits for all feedback (thanks @glozow @Xekyo @LarryRuane !).  Added a summary for libsecp256k1 1192 with the help of @apoelstra and @sipa (thanks!), added a topic page for ephemeral anchors, reviewed @bitschmidty's section (thanks!), removed the releases/RC section (none this week), and added the topic links.

Thanks everyone!

bitschmidty commented:
Squashed and merged! Thank you @harding @Xekyo @sipa @apoelstra for authoring and @LarryRuane @glozow @Xekyo for reviewing! 🚀
